sigstore/sign.py

```diff
@@ -229,6 +230,45 @@ def sign_dsse(
 
         return self._finalize_sign(cert, content, proposed_entry)
 
+    def sign_dsse_envelope(
```

Comment:
I'm a little wary of a new top-level sign API, especially one that's so close to `sign_dsse` -- instead of making this a new function, what do you think about making `sign_dsse` take `Statement | Envelope`?

From there, we'd need to improve the docs. But I think that might be a little cleaner/easier to maintain long-term 🙂